### ISSUE TITLE
[Bug] eslint-plugin-vue doesn't working with separate files

### ISSUE THREAD

**Issue by runyasak** [vuejs/eslint-plugin-vue#887]
**Tell us about your environment**	
* **ESLint version:** 5.16.0
* **eslint-plugin-vue version:** 5.0.0
* **Node version:**	 10.15.2

**Please show your full configuration:**

```
{
  root: true,
  env: {
    node: true
  },
  'extends': [
    'plugin:vue/recommended',
    '@vue/standard'
  ],
  rules: {
    'no-console': process.env.NODE_ENV === 'production' ? 'error' : 'off',
    'no-debugger': process.env.NODE_ENV === 'production' ? 'error' : 'off',
    'vue/html-closing-bracket-newline': 'off',
    'vue/html-self-closing': 'off'
  },
  parserOptions: {
    parser: 'babel-eslint'
  }
}
```

**What did you do?**
I have created a Vue single file with separating script file, like this [vue-separate-files](https://vuejs.org/v2/guide/single-file-components.html#What-About-Separation-of-Concerns). 
First, the script was separated and it didn't warn any sign. But, when I wrote them in .vue file, warnings are appeared.

```vue
// MyComponent.vue
<template>
  <div class="container">
    <p>{{ label }}</p>
  </div>
</template>

<script src="my-script.js"></script>
```

```js
// my-script.js
export default {
  name: 'MyComponent',
  props: {
    label: String
  }
}
```
It should appear `eslint(vue/require-default-prop)` but it doesn't.

**What did you expect to happen?**
The eslint warning should be worked on a js file.

**What actually happened?**

The eslint doesn't show any sign of warning.

**Comment by mysticatea** [comment]
Thank you for this issue.

But this is not a bug, because ESLint checks each file independently.

**Comment by AGPDev** [comment]
I have this problem, 4 hours trying resolve the same problem, and 'vue/order-in-components',

After see this answer i run test with clean Vue CLI 3 project, and using .vue the lint works but make import js in src script tag not work.

so @mysticatea YES, this probably a bug

**Comment by AGPDev** [comment]
@runyasak try add  `// @vue/component` in your .js see https://github.com/vuejs/vue-cli/issues/1602,

For me this work

**Comment by ghm-salvarado** [comment]
I came here looking for a solution for the same thing. @AGPDev got it right. 

Just wanted to add for anyone who finds this in the future, as explained in vuejs/vue-cli#1602, the `// @vue/component` marker needs to come immediately before `export default { ... }` in your script — **not necessarily at the top of the file**. I was putting it before my `import`s which doesn't work. Totally fine when you do it correctly.
